extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java

```diff
@@ -412,63 +426,80 @@ private boolean shouldActivateService(Capabilities capabilities,
 
-            Map<String, byte[]> files = WebJarUtil.production(curateOutcomeBuildItem, artifact, GRAPHQL_UI_WEBJAR_PREFIX);
+                for (Map.Entry<String, byte[]> file : files.entrySet()) {
 
-            for (Map.Entry<String, byte[]> file : files.entrySet()) {
+                    String fileName = file.getKey();
+                    byte[] content = file.getValue();
+                    if (fileName.endsWith(FILE_TO_UPDATE)) {
+                        content = WebJarUtil
+                                .updateUrl(new String(content, StandardCharsets.UTF_8), graphQLPath, LINE_TO_UPDATE,
+                                        LINE_FORMAT)
+                                .getBytes(StandardCharsets.UTF_8);
+                    }
+                    fileName = GRAPHQL_UI_FINAL_DESTINATION + "/" + fileName;
 
-                String fileName = file.getKey();
-                byte[] content = file.getValue();
-                if (fileName.endsWith(FILE_TO_UPDATE)) {
-                    content = WebJarUtil
-                            .updateUrl(new String(content, StandardCharsets.UTF_8), graphQLPath, LINE_TO_UPDATE, LINE_FORMAT)
-                            .getBytes(StandardCharsets.UTF_8);
+                    generatedResourceProducer.produce(new GeneratedResourceBuildItem(fileName, content));
+                    nativeImageResourceProducer.produce(new NativeImageResourceBuildItem(fileName));
                 }
-                fileName = GRAPHQL_UI_FINAL_DESTINATION + "/" + fileName;
 
-                generatedResourceProducer.produce(new GeneratedResourceBuildItem(fileName, content));
-                nativeImageResourceProducer.produce(new NativeImageResourceBuildItem(fileName));
+                smallRyeGraphQLBuildProducer.produce(new SmallRyeGraphQLBuildItem(GRAPHQL_UI_FINAL_DESTINATION,
+                        httpRootPath.adjustPath(quarkusConfig.ui.rootPath)));
             }
+        }
+    }
 
-            Handler<RoutingContext> handler = recorder
-                    .uiHandler(GRAPHQL_UI_FINAL_DESTINATION, httpRootPath.adjustPath(quarkusConfig.ui.rootPath));
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void registerGraphQLUiHandler(
+            BuildProducer<RouteBuildItem> routeProducer,
+            SmallRyeGraphQLRecorder recorder,
+            SmallRyeGraphQLRuntimeConfig runtimeConfig,
+            SmallRyeGraphQLBuildItem smallRyeGraphQLBuildItem) throws Exception {
+
+        if (shouldInclude()) {
+            Handler<RoutingContext> handler = recorder.uiHandler(smallRyeGraphQLBuildItem.getGraphqlUiFinalDestination(),
+                    smallRyeGraphQLBuildItem.getGraphqlUiPath(), runtimeConfig);
             routeProducer.produce(new RouteBuildItem(quarkusConfig.ui.rootPath, handler));
             routeProducer.produce(new RouteBuildItem(quarkusConfig.ui.rootPath + "/*", handler));
         }
     }
+
+    private boolean shouldInclude() {
```

Comment:
You should make this static and pass the parameters here.

In a perfect world, the config would also be injected in the build steps that require it.

Reply:
Ok, I can do that